### PR TITLE
If mountPath is missing, prefix with root dir.

### PR DESCRIPTION
**Pull request by brendandburns**
If `mountPath` is not absolute, add the os-specific root directory as a prefix.

Ref: https://github.com/kubernetes/kubernetes/pull/51240
https://groups.google.com/forum/#!topic/kubernetes-sig-storage/k_0Wr2kYkpU

@thockin @saad-ali @andyzhangx 

Note to @thockin I left the validation in place, in seems like it is a net win, since it will give a decent
error message to most people, but the fall-back defaulting is there if it doesn't catch 'c:'

I'm happy to rip out the validation if that is preferable to everyone.  Let me know.

```release-note
If a non-absolute mountPath is passed to the kubelet, prefix it with the appropriate root path.
```

### CODE REVIEW

File: pkg/kubelet/kubelet_pods.go [hunk 1]
```diff
@@ -111,6 +111,23 @@ func (kl *Kubelet) makeDevices(pod *v1.Pod, container *v1.Container) ([]kubecont
 	return devices, nil
 }
 
+func makeAbsolutePath(goos, path string) string {
```
Comment (thockin) [inline pkg/kubelet/kubelet_pods.go]: Should this just be in kubelet_windows.go and kubelet_linux.go files?

**Review by thockin — COMMENTED** [review]
Need to remove all the apiserver validation, too.

And update API docs (comments)

And regenerate all the crap

File: pkg/kubelet/kubelet_pods.go (hunk 1 above)
Comment (brendandburns) [reply pkg/kubelet/kubelet_pods.go]: I like it here better, since that way it will run in the unit tests on either platform.

**Comment by thockin** [comment]
??? Only part of it will run in unit tests on either patform.  There should
be no difference in LOCs-executed, just in where the code lives and whether
it is compiled in or not?

**Comment by brendandburns** [comment]
No, if you notice, the absolute path function takes the `goos` as a parameter, and I pass strings in the test, so I test most of the  `GOOS=windows` path even on Linux.

**Comment by brendandburns** [comment]
Also, can I get a decision between @thockin and @saad-ali about whether or not to rip out the validation?

**Comment by thockin** [comment]
If we keep the validation, it fires most of the time, but sometimes wrongly.

I think it's simpler just to remove it.  @saad-ali - do you disagree?

**Comment by brendandburns** [comment]
fwiw, it never fires wrongly. it sometimes _doesn't_ fire when it should...

Again, don't feel strongly either way, just looking for consensus.

**Comment by brendandburns** [comment]
Friendly ping on this. Given @thockin's desire to get this in for 1.9, I need a quick consensus...

thanks
--brendan

**Comment by saad-ali** [comment]
> I think it's simpler just to remove it.  @saad-ali - do you disagree?

I agree, let's pull it out.

**Comment by brendandburns** [comment]
ok, will do.

**Comment by brendandburns** [comment]
Validation removed, documentation updated, please take another look...

I tried to regenerate things, but none of the scripts I ran seemed to actually make any changes.

I guess I'll see if automated unit tests pass, and fix any problems that might occur...

**Comment by saad-ali** [comment]
Unit test needs to be updated for removal of validation:

```
--- FAIL: TestValidateVolumeMounts (0.00s)
	validation_test.go:3862: expected failure for windows absolute subpath
```

Otherwise LGTM

**Comment by brendandburns** [comment]
@saad-ali test fixed (that's what I get for not re-running tests after a rebase *sigh*)

ptal, thanks!

**Comment by brendandburns** [comment]
@thockin tests are passing, @saad-ali LGTM'd any further comments?

Thanks
--brendan

**Comment by dims** [comment]
LGTM 👍

**Comment by andyzhangx** [comment]
@brendanburns so will this PR cherry-pick to v1.8? I can help do that.

**Pull request merged**